### DISCUSSION TITLE
Google Cloud Managed Service for Prometheus

### DISCUSSION THREAD

**Discussion by DazWilkin** [prometheus/promlens#78] [Show and tell]
In order for PromLens to work with [Google Cloud Managed Service for Prometheus (GMP)](https://cloud.google.com/stackdriver/docs/managed-prometheus), (Prometheus) API requests must be authenticated (and the identity must be authorized).

I hacked together a basic proxy ([`promlens-gmp-token-proxy`](https://github.com/DazWilkin/promlens-gmp-token-proxy)) that should<sup>1</sup> be run on same host as PromLens, functions as a proxy including adding an `Authorization` header to PromLens requests to authenticate to GMP.

There's an explanation here: [Authenticating PromLens to Google Managed Prometheus](https://pretired.dazwilkin.com/posts/230104/)

<sup>1</sup> -- For simplicity the proxy provides a non-TLS (!) HTTP endpoint to PromLens even though it uses TLS to make requests to GMP.

**Comment by holmesb** [comment]
Hi @DazWilkin, not sure if you wrote this and your blog prior to Google creating\documenting their own [frontend proxy](https://cloud.google.com/stackdriver/docs/managed-prometheus/query#ui-prometheus)? Pointing PromLens at URL of frontend proxy (which has authn token mounted) seems to allow it to access data fine. Can auto-complete metrics and graph them, but the table tab errors ‘_Error: Missing “data” field in response JSON_’: 

![image](https://github.com/prometheus/promlens/assets/5072156/0d4ba901-3134-4611-9f8d-398ef436647e)

Have you ever hit this error? Debug logs show nothing. On promlens v0.3.0.

**Reply by DazWilkin** [reply]
Are you able to determine the underlying API call using the browser's DevTools?

I've not looked at this for some time, but I revisited https://demo.promlens.com/ and, `Table` uses `/api/v1/query` whereas e.g. `Graph` uses `/api/v1/query_range`.

PromLens occasionally adds `refresh` parameter to the query string for `/api/v1/query` and GMP (used to!?) balk at this (Google IssueTracker Issue [264655965](https://issuetracker.google.com/issues/))
